Monika After Story/game/zz_apikeys.rpy

```diff
@@ -0,0 +1,556 @@
+            pass
+
+        def action_ok(self):
+            """
+            Called when the ok button is clicked
+            """
+            self.action_cancel()
+
+        def action_retry(self):
+            """
+            Called when the retry button is clicked
+            """
+            self.start()
+
+        def _hide_all_buttons(self):
+            """
+            Hides all buttons
+            """
+            self._show_cancel = False
+            self._show_ok = False
+            self._show_retry = False
+
+        def _set_buttons_ok(self):
+            """
+            Sets button flags for ok available
+            """
+            self._hide_all_buttons()
+            self._show_ok = True
+
+        def _set_buttons_retrycancel(self):
+            """
+            Sets button flags for retrycancel available
+            """
+            self._hide_all_buttons()
+            self._show_cancel = True
+            self._show_retry = True
+
+        def _set_state_checking_update(self):
+            """
+            Sets text and button flags for when an update is in progress.
+            This includes checking for an update.
+            """
+            self._main_text = "Updating cert{0} (This can take a while)".format(
+                "." * (self._loop_counter % 4)
+            )
+            self._hide_all_buttons()
+            self._show_cancel = True
+
+        def _set_state_error(self):
+            """
+            Sets text and button flags for when update failed 
+            """
+            if mas_can_import.certifi():
+                import certifi
+                
+                if self._rv_value == certifi.RV_ERR_CERT_WRITE:
+                    self._main_text = (
+                        "ERROR: failed to write cert to disk. Try again later "
+                        "or manually update the cert."
+                    )
+                    self._set_buttons_retrycancel()
+
+                elif self._rv_value == certifi.RV_ERR_BAD_SSL_LIB:
+                    self._main_text = (
+                        "ERROR: SSL is not available. Cannot update cert"
+                    )
+                    self._set_buttons_ok()
+
+                else:
+                    self._main_text = (
+                        "ERROR: failed to update cert. Try again later or "
+                        "manually update the cert."
+                    )
+                    self._set_buttons_retrycancel()
+
+            else:
+                self._main_text = (
+                    "ERROR: certifi not available. Cannot update cert"
+                )
+                self._set_buttons_ok()
+
+        def _set_state_no_update(self):
+            """
+            Sets text and button flags for when no update is found
+            """
+            self._main_text = "Cert already up to date!"
+            self._hide_all_buttons()
+            self._show_ok = True
+            self._show_retry = True
```

Comment:
I think we shouldn't allow retry, it was a successful request already.

Reply:
What do you mean by "successful request"? In the loop code, "successful" means the cert was updated, in this case, no update was found, but that could be because the user's cert was up to date, or the server that we're pulling from hasn't updated yet. Providing an option to retry can handle the latter case.

Reply:
I just never saw a "try again" button when an app tells me it's up to date. Like how our updater doesn't allow to check again if it knows the mod is up to date.

Reply:
ok, retry removed